### PR TITLE
fix!: improve user experience on setting compression type

### PR DESCRIPTION
**Pull request by lyang24**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?
better user experience to set the compression type in all lower case
- rename all within CompressionType enum to lowercase 

## Checklist

- [x]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)
fixes: https://github.com/GreptimeTeam/greptimedb/issues/2758

### CODE REVIEW

File: src/common/datasource/src/compression.rs [hunk 1]
```diff
@@ -27,6 +27,7 @@ use tokio_util::io::{ReaderStream, StreamReader};
 
 use crate::error::{self, Error, Result};
 #[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, EnumIter, Serialize, Deserialize)]
+#[serde(rename_all = "lowercase")]
```
Comment (lyang24) [inline src/common/datasource/src/compression.rs:30]: I would use this for deser only `#[serde(rename_all(deserialize = "lowercase"))]` but would this create a problem when we serialize then deserialize from serialized result?

Comment (evenyag) [reply src/common/datasource/src/compression.rs:30]: Yes. Tried in the playground, `serde` can't deserialize the json string:

https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=63a3955bc5529951b97db199fdc6bc14

File: src/common/datasource/src/compression.rs [hunk 2]
```diff
@@ -27,6 +27,7 @@ use tokio_util::io::{ReaderStream, StreamReader};
 
 use crate::error::{self, Error, Result};
 #[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, EnumIter, Serialize, Deserialize)]
```
Comment (evenyag) [inline src/common/datasource/src/compression.rs:29]:
```suggestion
use crate::error::{self, Error, Result};

#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, EnumIter, Serialize, Deserialize)]
```

**Review by evenyag — APPROVED** [review]
Rest LGTM.

**Review by waynexia — APPROVED** [review]

**Pull request merged**